lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -112,12 +112,24 @@ export default function LMSFilePicker({
           path: listFilesApi.path,
         })
       );
+
+      /**
+       * FIXME: This is a temporary fix for the updated file API response for
+       * BlackBoard files. A File returned by the BlackBoard files API may be
+       * either a `File` or a `Folder`. Until `FileList` and this component
+       * have has been updated to render and handle `Folder`-type objects,
+       * filter them out of the file list: i.e. hide them.
+       */
+      const filteredFiles = files.filter(
+        file => !file.type || file.type === 'File'
```

Comment:
Is the `type` property optional because it isn't included in the response for Canvas, or is it in fact always set? It would be good to verify if this inconsistency exists and if so eliminate it in future.

Reply:
> Is the type property optional because it isn't included in the response for Canvas...

My _assumption_ was that `type` is not present on Canvas responses, that the new properties—`type`, `contents`, `parent_id`—are only present in responses from the Blackboard proxy API. But you are correct: assumption is no good. 

This code is plenty safe as-is, so I'm going to take this question as a follow-up task; adding it to my TODO list.